### PR TITLE
Docs/overview and link

### PR DESCRIPTION
**Pull request by apenzk**
# Description of change

Merged wiki documentation with docs documentation. There is danger for duplication if the documentation is kept in two different places. Once this is merged we need to clean up the wiki itself (i.e. remove the wiki files from there)

## Type of change

- Documentation Fix

## How the change has been tested

No testing.

## Change checklist

- [ ] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/OVERVIEW.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+- [Application logic](./application_logic.md)
+  - [Protocol high level overview](./application_logic/protocol.md)
+- [Implementation design](./implementation_design.md)
+  - [Event driven model](./implementation_design/event_driven_model.md)
+  - [Packages and plugins](./implementation_design/packages_plugins.md)
+  - [Configuration parameters](./implementation_design/configuration_parameters.md)
+  - [Plugin](./implementation_design/plugin.md)
+  - [Object storage](./implementation_design/object_storage.md)
+- [API](./apis/api.md)
+  - [Client Lib: Interaction with Layers](./clientLib.md)
```
Comment (jkrvivian) [inline docs/OVERVIEW.md:26]:
```suggestion
  - [Client Lib: Interaction with Layers](./apis/clientLib.md)
```

File: docs/concepts/glossary.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
+#### TANGLE
+An append only data structure where each item references two other items.
+
+#### MESSAGE
+The object type which is gossiped between neighbors. All gossiped information is included in a message.
+
+#### MESSAGE TANGLE
+The collection of all messages.
+
+#### OBJECT
```
[the remark this answers is not in the document]
Comment (apenzk) [reply docs/concepts/glossary.md:10]: mhm but everything in this doc is described by object (value object, FPC object, etc.).
By what has it been replaced?

File: docs/concepts/glossary.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
+#### TANGLE
+An append only data structure where each item references two other items.
+
+#### MESSAGE
```
[the remark this answers is not in the document]
Comment (apenzk) [reply docs/concepts/glossary.md:4]: i added these.

File: docs/concepts/glossary.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+A field in an object which can only be filled by another object.
+
+#### VALUE TANGLE
+The collection of all value objects.
+
+#### VALUE TRANSFER APPLICATION
+The application which maintains the ledger state.
+
+#### VALUE OBJECT
+The basic object of the value transfer application.
```
[the remark this answers is not in the document]
Comment (apenzk) [reply docs/concepts/glossary.md:34]: ok done

File: docs/concepts/glossary.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+#### TRANSACTION
+The payload of a value object. It contains the particulars of a transfer of funds.
```
[the remark this answers is not in the document]
Comment (apenzk) [reply docs/concepts/glossary.md:37]: of a value message?

File: docs/tutorials/custom_dRNG.md [hunk 6]
```diff
@@ -0,0 +1 @@
+All the steps are described in the [dRNG wiki](https://github.com/iotaledger/drng/wiki).
```
[the remark this answers is not in the document]
Comment (apenzk) [reply docs/tutorials/custom_dRNG.md:1]: this is not the goshimmer wiki but the drng wiki

File: docs/concepts/glossary.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (apenzk) [reply docs/concepts/glossary.md]: i addressed this now. it should be ok.

File: docs/concepts/glossary.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
+#### TANGLE
+An append only data structure where each item references two other items.
```
[the remark this answers is not in the document]
Comment (apenzk) [reply docs/concepts/glossary.md]: ok i adressed this.

File: docs/concepts/glossary.md [hunk 8]
```diff
@@ -22,7 +22,7 @@ A field in a message which determines the type. Examples are:
 The application which maintains the ledger state.
 
 #### TRANSACTION
-The payload of a value message. It contains the particulars of a transfer of funds.
+A message with payload of type value. It contains the information of a transfer of funds.
```
Comment (acha-bill) [inline docs/concepts/glossary.md:25]: Transaction payload type is `TransactionType`

File: docs/concepts/glossary.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+#### OBJECT
+The most basic unit of information of the IOTA Protocol. Each object has a type and size and contains data.
+
+#### CORE OBJECT TYPE
+An object type which must be parsed by all users. Core object types include:
+* Value objects
+* FPC Opinion objects
+* DRNG objects
+* Salt declaration objects
+* Generic data objects
```
[the remark this answers is not in the document]
Comment (apenzk) [reply docs/concepts/glossary.md]: done

File: docs/concepts/glossary.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (apenzk) [reply docs/concepts/glossary.md]: resolved.

File: docs/concepts/glossary.md (hunk 8 above)
Comment (apenzk) [reply docs/concepts/glossary.md:25]: thx i have added this info. Also in the the glossary point PAYLOAD i have added the type where applicable. 
E.g. For FPC there is a StatementType. However for faucet and the syncbeacon the name is more generic ("payloadType"). We should make the later more specific in a different PR.

File: docs/apis/clientLib.md [hunk 10]
```diff
@@ -0,0 +1,439 @@
...
+Optionally, define your own `http.Client` to use, in order for example to define custom timeouts:
+```
+goshimAPI := client.NewGoShimmerAPI("http://mynode:8080", http.Client{Timeout: 30 * time.Second})
+```
+
+#### A note about errors
+
+The API issues HTTP calls to the defined GoShimmer node. Non 200 HTTP OK status codes will reflect themselves as `error` in the returned arguments. Meaning that for example calling for attachments with a non existing/available transaction on a node, will return an `error` from the respective function. (There might be exceptions to this rule)
+
+# Communication Layer APIs
```
[the remark this answers is not in the document]
Comment (apenzk) [reply docs/apis/clientLib.md]: I have rearanged the files now.

File: docs/OVERVIEW.md [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+- [Implementation design](./implementation_design.md)
+  - [Event driven model](./implementation_design/event_driven_model.md)
+  - [Packages and plugins](./implementation_design/packages_plugins.md)
+  - [Configuration parameters](./implementation_design/configuration_parameters.md)
+  - [Plugin](./implementation_design/plugin.md)
+  - [Object storage](./implementation_design/object_storage.md)
+- [API](./apis/api.md)
+  - [WebAPI](./apis/webAPI.md)
+  - [Mana](./apis/mana.md)
+  - [Mana (deprecated)](./apis/mana_old.md)
```
Comment (jkrvivian) [inline docs/OVERVIEW.md:27]: Remove this.

File: docs/apis/mana_old.md [hunk 12]
```diff
@@ -0,0 +1,236 @@
+# Mana APIs (deprecated)
```
Comment (jkrvivian) [inline docs/apis/mana_old.md:1]: Let's remove this file, everything here is in the mana.md.

File: docs/OVERVIEW.md (hunk 11 above)
Comment (apenzk) [reply docs/OVERVIEW.md]: done

File: docs/apis/mana_old.md (hunk 12 above)
Comment (apenzk) [reply docs/apis/mana_old.md]: ok. removed it

**Review by jkrvivian — APPROVED** [review]

**Pull request merged**